SPR not resolving repository name

I have installed SPR with use of deb package. This is the installed version:
```spr version : 0.2.0 : 2021-05-27T05:05:34Z : c49a3038```.

When trying to check the PRs status I get:
```bash
$ git spr --status
panic: Could not resolve to a Repository with the name '/'.

goroutine 1 [running]:
github.com/ejoffe/spr/spr.check(...)
        /Users/eitan/Code/spr/spr/spr.go:788
github.com/ejoffe/spr/spr.(*stackediff).getGitHubInfo(0xc0000c0480, 0x7d5ad0, 0xc0000a8000, 0xc0000bc050, 0x747280)
        /Users/eitan/Code/spr/spr/spr.go:489 +0xa15
github.com/ejoffe/spr/spr.(*stackediff).StatusPullRequests(0xc0000c0480, 0x7d5ad0, 0xc0000a8000)
        /Users/eitan/Code/spr/spr/spr.go:248 +0x92
main.main()
        /Users/eitan/Code/spr/cmd/spr/main.go:74 +0x3ab
```
This is worrying for two reasons:
1. Hardcoded(?) paths
2. Inability to find the repo name.

For 2. I checked the config file:
```yaml
$ cat .spr.yml 
githubRepoOwner: ""
githubRepoName: ""
requireChecks: true
requireApproval: true
showPRLink: true
```

After putting repo owner and repo name manually, the panic is not occurring.

--- a/spr/config.go
+++ b/spr/config.go
@@ -83,10 +83,11 @@ func defaultConfig() *Config {
 }
 
 func getRepoDetailsFromRemote(remote string) (string, string, bool) {
-	regex := regexp.MustCompile(`^origin\s+https://github.com/(\w+)/(\w+).git \(push\)`)
+	regex := regexp.MustCompile(`^origin\s+(?:(https://)|(ssh://))?(git@)?github.com/(\w+)/(\w+)(.git)? \(push\)`)
 	matches := regex.FindStringSubmatch(remote)
 	if matches != nil {
-		return matches[1], matches[2], true
+		fmt.Printf("Matches: %q\n", matches)
+		return matches[4], matches[5], true
 	}
 	return "", "", false
 }

--- a/spr/config.go
+++ b/spr/config.go
@@ -83,11 +83,23 @@ func defaultConfig() *Config {
 }
 
 func getRepoDetailsFromRemote(remote string) (string, string, bool) {
-	regex := regexp.MustCompile(`^origin\s+(?:(https://)|(ssh://))?(git@)?github.com/(\w+)/(\w+)(.git)? \(push\)`)
+	// Allows "https://", "ssh://" or no protocol at all (this means ssh)
+	protocolFormat := `(?:(https://)|(ssh://))?`
+	// This may or may not be present in the address
+	userFormat := `(git@)?`
+	// "/" is expected in "http://" or "ssh://" protocol, when no protocol given
+	// it should be ":"
+	repoFormat := `github.com(/|:)(?P<repoOwner>\w+)/(?P<repoName>\w+)`
+	// This is neither required in https access nor in ssh one
+	suffixFormat := `(.git)?`
+	regexFormat := fmt.Sprintf(`^origin\s+%s%s%s%s \(push\)`,
+		protocolFormat, userFormat, repoFormat, suffixFormat)
+	regex := regexp.MustCompile(regexFormat)
 	matches := regex.FindStringSubmatch(remote)
 	if matches != nil {
-		fmt.Printf("Matches: %q\n", matches)
-		return matches[4], matches[5], true
+		repoOwnerIndex := regex.SubexpIndex("repoOwner")
+		repoNameIndex := regex.SubexpIndex("repoName")
+		return matches[repoOwnerIndex], matches[repoNameIndex], true
 	}
 	return "", "", false
 }

--- a/spr/config.go
+++ b/spr/config.go
@@ -61,14 +61,12 @@ func defaultConfig() *Config {
 	mustgit("remote -v", &output)
 	lines := strings.Split(output, "\n")
 
-	regex := regexp.MustCompile(`^origin\s+https://github.com/(\w+)/(\w+).git \(push\)`)
-
 	for _, line := range lines {
-		matches := regex.FindStringSubmatch(line)
-		if matches != nil {
+		repoOwner, repoName, match := getRepoDetailsFromRemote(line)
+		if match {
 			return &Config{
-				GitHubRepoOwner: matches[1],
-				GitHubRepoName:  matches[2],
+				GitHubRepoOwner: repoOwner,
+				GitHubRepoName:  repoName,
 				RequireChecks:   true,
 				RequireApproval: true,
 				ShowPRLink:      true,
@@ -82,6 +80,15 @@ func defaultConfig() *Config {
 		RequireApproval: true,
 		ShowPRLink:      true,
 	}
+}
+
+func getRepoDetailsFromRemote(remote string) (string, string, bool) {
+	regex := regexp.MustCompile(`^origin\s+https://github.com/(\w+)/(\w+).git \(push\)`)
+	matches := regex.FindStringSubmatch(remote)
+	if matches != nil {
+		return matches[1], matches[2], true
+	}
+	return "", "", false
 }
 
 func writeConfigFile(filename string, config *Config) {

--- a/spr/config_test.go
+++ b/spr/config_test.go
@@ -20,12 +20,12 @@ func TestGetRepoDetailsFromRemote(t *testing.T) {
 		{"origin  ssh://git@github.com/r2/d2.git (fetch)", "", "", false},
 		{"origin  ssh://git@github.com/r2/d2 (push)", "r2", "d2", true},
 
-		{"origin  git@github.com/r2/d2.git (push)", "r2", "d2", true},
-		{"origin  git@github.com/r2/d2.git (fetch)", "", "", false},
-		{"origin  git@github.com/r2/d2 (push)", "r2", "d2", true},
+		{"origin  git@github.com:r2/d2.git (push)", "r2", "d2", true},
+		{"origin  git@github.com:r2/d2.git (fetch)", "", "", false},
+		{"origin  git@github.com:r2/d2 (push)", "r2", "d2", true},
 	}
 	for i, testCase := range testCases {
-		t.Logf("Testing %v %v", i, testCase.remote)
+		t.Logf("Testing %v %q", i, testCase.remote)
 		repoOwner, repoName, match := getRepoDetailsFromRemote(testCase.remote)
 		if repoOwner != testCase.repoOwner {
 			t.Fatalf("Wrong \"repoOwner\" returned for test case %v, expected %q, got %q", i, testCase.repoOwner, repoOwner)

--- a/spr/config_test.go
+++ b/spr/config_test.go
@@ -14,18 +14,27 @@ func TestGetRepoDetailsFromRemote(t *testing.T) {
 	testCases := []testCase{
 		{"origin  https://github.com/r2/d2.git (push)", "r2", "d2", true},
 		{"origin  https://github.com/r2/d2.git (fetch)", "", "", false},
+		{"origin  https://github.com/r2/d2 (push)", "r2", "d2", true},
+
+		{"origin  ssh://git@github.com/r2/d2.git (push)", "r2", "d2", true},
+		{"origin  ssh://git@github.com/r2/d2.git (fetch)", "", "", false},
+		{"origin  ssh://git@github.com/r2/d2 (push)", "r2", "d2", true},
+
+		{"origin  git@github.com/r2/d2.git (push)", "r2", "d2", true},
+		{"origin  git@github.com/r2/d2.git (fetch)", "", "", false},
+		{"origin  git@github.com/r2/d2 (push)", "r2", "d2", true},
 	}
 	for i, testCase := range testCases {
 		t.Logf("Testing %v %v", i, testCase.remote)
 		repoOwner, repoName, match := getRepoDetailsFromRemote(testCase.remote)
 		if repoOwner != testCase.repoOwner {
-			t.Fatalf("Wrong \"repoOwner\" returned for test case %v", i)
+			t.Fatalf("Wrong \"repoOwner\" returned for test case %v, expected %q, got %q", i, testCase.repoOwner, repoOwner)
 		}
 		if repoName != testCase.repoName {
-			t.Fatalf("Wrong \"repoName\" returned for test case %v", i)
+			t.Fatalf("Wrong \"repoName\" returned for test case %v, expected %q, got %q", i, testCase.repoName, repoName)
 		}
 		if match != testCase.match {
-			t.Fatalf("Wrong \"match\" returned for test case %v", i)
+			t.Fatalf("Wrong \"match\" returned for test case %v, expected %t, got %t", i, testCase.match, match)
 		}
 	}
 }

--- a/spr/config_test.go
+++ b/spr/config_test.go
@@ -1,0 +1,31 @@
+package spr
+
+import (
+	"testing"
+)
+
+func TestGetRepoDetailsFromRemote(t *testing.T) {
+	type testCase struct {
+		remote    string
+		repoOwner string
+		repoName  string
+		match     bool
+	}
+	testCases := []testCase{
+		{"origin  https://github.com/r2/d2.git (push)", "r2", "d2", true},
+		{"origin  https://github.com/r2/d2.git (fetch)", "", "", false},
+	}
+	for i, testCase := range testCases {
+		t.Logf("Testing %v %v", i, testCase.remote)
+		repoOwner, repoName, match := getRepoDetailsFromRemote(testCase.remote)
+		if repoOwner != testCase.repoOwner {
+			t.Fatalf("Wrong \"repoOwner\" returned for test case %v", i)
+		}
+		if repoName != testCase.repoName {
+			t.Fatalf("Wrong \"repoName\" returned for test case %v", i)
+		}
+		if match != testCase.match {
+			t.Fatalf("Wrong \"match\" returned for test case %v", i)
+		}
+	}
+}

